packages/auth/src/authSession.ts

```diff
@@ -107,59 +222,196 @@ export class AuthSession {
   ): Promise<OauthProviderInfo> {
     this.throwIfDisconnected();
     return notImplemented(params);
   }
 
+  /**
+   * Removes an OAuth provider from this user's authentication methods.
+   *
+   * @param {string} providerId - The ID of the OAuth provider to remove
+   * @returns {Promise<void>} A promise that resolves when the provider is removed
+   */
   public async removeOauthProvider(providerId: string): Promise<void> {
     this.throwIfDisconnected();
     return notImplemented(providerId);
   }
 
+  /**
+   * Adds a new passkey (WebAuthn credential) to this user's authentication methods.
+   *
+   * @param {CredentialCreationOptions} params - The credential creation options for WebAuthn
+   * @returns {Promise<PasskeyInfo>} A promise that resolves to the created passkey info
+   */
   public async addPasskey(
     params: CredentialCreationOptions,
   ): Promise<PasskeyInfo> {
     this.throwIfDisconnected();
     return notImplemented(params);
   }
 
+  /**
+   * Removes a passkey (WebAuthn credential) from this user's authentication methods.
+   *
+   * @param {string} authenticatorId - The ID of the authenticator/passkey to remove
+   * @returns {Promise<void>} A promise that resolves when the passkey is removed
+   */
   public async removePasskey(authenticatorId: string): Promise<void> {
     this.throwIfDisconnected();
     return notImplemented(authenticatorId);
   }
 
+  /**
+   * Disconnects and invalidates this authentication session.
+   *
+   * This method marks the session as disconnected and clears any stored
+   * credentials in the Turnkey stamper.
+   *
+   * @returns {Promise<void>} A promise that resolves when the session is disconnected
+   */
   public async disconnect(): Promise<void> {
     this.isDisconnected = true;
     (this.turnkey.stamper as TurnkeyStamper).clear?.();
   }
 
+  /**
+   * Serializes the current authentication session state to a JSON string.
+   *
+   * This method creates a serializable representation of the session that includes
+   * user information, authentication type, credential bundles (for non-passkey auth),
+   * credential IDs (for passkey auth), and expiration time.
+   *
+   * The serialized state can be stored and later used to restore the session
+   * using AuthClient.loadAuthSessionState().
+   *
+   * @returns {string} A JSON string containing the serialized session state
+   *
+   * @example
+   * ```ts twoslash
+   * const sessionState = authSession.getAuthSessionState();
+   * localStorage.setItem('authSession', sessionState);
+   *
+   * // Later restore:
+   * const savedState = localStorage.getItem('authSession');
+   * if (savedState) {
+   *   const restoredSession = await authClient.loadAuthSessionState(JSON.parse(savedState));
+   * }
+   * ```
+   */
+  public getAuthSessionState(): string {
+    this.throwIfDisconnected();
+
+    // Calculate expiration time (24 hours from now as default)
+    const expirationDateMs = Date.now() + 24 * 60 * 60 * 1000;
```

Comment:
We're going to want to update this later, right? The expiration time won't always be 24 hours from the time the state is serialized. We can do that when we add expiration handling in general, but I might add a TODO.